drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/editor/menu/MenuItemView.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2018 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.drools.workbench.screens.scenariosimulation.client.editor.menu;
+
+import com.google.gwt.dom.client.LIElement;
+import com.google.gwt.event.dom.client.ClickEvent;
+import com.google.gwt.user.client.Command;
+
+public interface MenuItemView {
+
+    interface Presenter {
+
+        void executeCommand(LIElement clickedElement);
+
+        LIElement getLIElement(String id, String innerText, Command command);
+
+        void enableElement(boolean toEnable);
+
+        void onClickEvent(ClickEvent event);
```

Comment:
There is a lot of gwt classes passed for the Presenter. Presenter should not know about them.

Reply:
@Rikkola I did this modification because probably I misunderstood when you said that every "logic"/status change should be done in the presenter.
This is the implementation:

public void onClickEvent(ClickEvent event) {
        event.preventDefault();
        event.stopPropagation();
    }

Originally it was inside MenuItemViewImpl - I moved that later to adapt to guidelines.
So, I need to understand better, then. 😄

Reply:
I think the code your basing this work on is not pure MVP and that is what is causing this confusion. So this might have to be left as it is.